src/actions.t.sol

```diff
@@ -1,159 +1,242 @@
-        // approve nft transfer to proxy 
+    function issue(uint256 tokenId) public returns (uint256) {
+        assertEq(collateralNFT.ownerOf(tokenId), borrower_);
+        // approve nft transfer to proxy
         collateralNFT.approve(borrowerProxy_, tokenId);
-        bytes memory data = abi.encodeWithSignature("transferIssue(address,address,uint256)", address(shelf), address(collateralNFT), tokenId);
-        bytes memory response = borrowerProxy.execute(actions, data);
-        (uint loan) = abi.decode(response, (uint));
+        bytes memory data = abi.encodeWithSignature(
+            "transferIssue(address,address,uint256)", address(shelf), address(collateralNFT), tokenId
+        );
+        bytes memory response = borrowerProxy.userExecute(actions, data);
+        (uint256 loan) = abi.decode(response, (uint256));
         // assert: nft transferred to borrowerProxy
         assertEq(collateralNFT.ownerOf(tokenId), borrowerProxy_);
         // assert: loan created and owner is borrowerProxy
         assertEq(title.ownerOf(loan), borrowerProxy_);
         return loan;
     }
 
-   function testIssueLockBorrow() public {
-       // Borrower: Issue Loan
-       (uint tokenId, ) = issueNFT(borrower_);
-       uint loan = issue(tokenId);
-       uint price = 50;
-       uint amount = 25;
-       uint riskGroup = 2;
-
-       // Lender: lend
-       defaultInvest(100 ether);
-       hevm.warp(block.timestamp + 1 days);
-       coordinator.closeEpoch();
-
-       // Admin: set loan parameters
-       priceNFTandSetRisk(tokenId, price, riskGroup);
-
-       // Borrower: Lock & Borrow
-       borrowerProxy.execute(actions, abi.encodeWithSignature("lockBorrowWithdraw(address,uint256,uint256,address)", address(shelf), loan, amount, borrower_));
-       assertEq(collateralNFT.ownerOf(1), address(shelf));
-       // check if borrower received loan amount
-       assertEq(currency.balanceOf(borrower_), amount);
-   }
+    function testIssueLockBorrow() public {
+        // Borrower: Issue Loan
+        (uint256 tokenId,) = _issueNFT(borrower_);
+        uint256 loan = issue(tokenId);
+        uint256 price = 50;
+        uint256 amount = 25;
+        uint256 riskGroup = 0;
+
+        // Lender: lend
+        defaultInvest(100 ether);
+        vm.warp(block.timestamp + 1 days);
+        coordinator.closeEpoch();
+
+        // Admin: set loan parameters
+        priceNFTandSetRisk(tokenId, price, riskGroup);
+
+        // Borrower: Lock & Borrow
+        borrowerProxy.userExecute(
+            actions,
+            abi.encodeWithSignature(
+                "lockBorrowWithdraw(address,uint256,uint256,address)", address(shelf), loan, amount, borrower_
+            )
+        );
+        assertEq(collateralNFT.ownerOf(tokenId), address(shelf));
+        // check if borrower received loan amount
+        assertEq(currency.balanceOf(borrower_), amount);
+    }
 
     function testFailIssueLockBorrowerWithdrawCeilingNotSet() public {
```

Comment:
Not necessary for this PR, but just leaving a reminder that ideally we should rewrite these to use `vm.expectRevert` at some point.

Reply:
Yes, that's a good point